feat: Add work queue and context generation systems (Phase 3.1 - PR 1.3)

TimeToBuildBob opened the pull request:
## Summary

Adds work queue management and context generation infrastructure to gptme-agent-template.

## Work Queue System (state/)

**Manual Queue** (queue-manual.md):
- Template for manually maintained work queue
- Rich session context and strategic reasoning
- Updated during autonomous operation

**Generated Queue** (queue-generated.md):
- Template for auto-generated queue from tasks/GitHub
- Refreshed periodically for objective priorities
- Fallback when manual queue not maintained

**Documentation** (state/README.md):
- Complete guide to two-queue system
- CASCADE workflow integration
- Best practices and examples

## Scripts Documentation (scripts/)

**scripts/README.md**:
- Context generation system documentation
- Task management (tasks.py) when available
- Autonomous run infrastructure overview
- Search and utility scripts
- Pre-commit hooks

## Main README Updates

- Document state/ and scripts/ directories
- Explain work queue and context generation systems
- Link to detailed documentation
- Update workspace structure section

## Fixes

- Remove broken tasks.py symlink (will be added in PR 2.3 with gptme-contrib)
- Update documentation to note tasks.py as optional
- Fix broken markdown link in ARCHITECTURE.md

## Part of Phase 3.1

This is PR 1.3 in the gptme-agent-template sync:
- PR 1.1: Autonomous run infrastructure (merged) ✅
- PR 1.2: Lessons system structure (PR #28) ✅
- **PR 1.3: Work queue + context** (this PR) ⬅️

See ErikBjare/bob#167 for complete context.
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Introduces work queue and context generation systems, updates documentation, and fixes minor issues in the gptme-agent-template.
> 
>   - **Work Queue System**:
>     - Adds `queue-manual.md` for manually maintained work queue with strategic context.
>     - Adds `queue-generated.md` for auto-generated queue from tasks and GitHub.
>     - Updates `state/README.md` with a guide to the two-queue system and CASCADE workflow integration.
>   - **Scripts Documentation**:
>     - Adds `scripts/README.md` documenting context generation, task management, and autonomous run infrastructure.
>     - Updates `context.sh` to handle optional `tasks.py` for task status.
>   - **Main README Updates**:
>     - Documents `state/` and `scripts/` directories.
>     - Explains work queue and context generation systems.
>     - Updates workspace structure section.
>   - **Fixes**:
>     - Removes broken `tasks.py` symlink.
>     - Updates documentation to note `tasks.py` as optional.
>     - Fixes broken markdown link in `ARCHITECTURE.md`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=gptme%2Fgptme-agent-template&utm_source=github&utm_medium=referral)<sup> for beacef49f2394df9a74f8d1074f4e9cdf54f9296. You can [customize](https://app.ellipsis.dev/gptme/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

ErikBjare commented on .pre-commit-config.yaml:
```diff
@@ -43,13 +43,5 @@ repos:
-        language: system
-        types: [markdown]
-        pass_filenames: false
-        #files: ^tasks/.*\.md$
```
This should be included, or silently pass/skip if tasks.py doesn't exist/broken symlink.

ErikBjare commented:
@greptileai review

TimeToBuildBob commented:
✅ Addressed review feedback on optional tasks.py handling

**Changes made:**
1. **scripts/context.sh line 32**: Changed from `-f` (file exists) to `-x` (executable) check for robustness
2. **.pre-commit-config.yaml line 55**: Added validation hook that gracefully handles missing tasks.py with informative skip message
3. **README.md line 88**: Clarified that tasks.py is optional from gptme-contrib

The validation hook now:
- Runs `tasks.py check` if tasks.py is present and executable
- Prints friendly message and continues if not found (as it's optional)
- Follows same pattern as context.sh for consistency

Ready for re-review! 🚀
